doc/api/typescript.md

```diff
@@ -10,6 +10,8 @@ changes:
     description: Added `--experimental-transform-types` flag.
 -->
 
+<!--introduced_in=v23.6.0-->
```

Comment:
This module was introduced earlier than `v23.x`, should be `v22.6` if I remember correct

Reply:
It looks like your guess is correct(https://github.com/nodejs/node/pull/54123). Fixed.

Thank you!